// @grove/react-font-awesome/lib/FontAwesome.js
'use strict';

var _objectWithoutProperties = require('babel-runtime/helpers/object-without-properties')['default'];

var _extends = require('babel-runtime/helpers/extends')['default'];

var _Object$keys = require('babel-runtime/core-js/object/keys')['default'];

var _String$fromCodePoint = require('babel-runtime/core-js/string/from-code-point')['default'];

var _interopRequireDefault = require('babel-runtime/helpers/interop-require-default')['default'];

Object.defineProperty(exports, '__esModule', {
  value: true
});

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _radium = require('radium');

var _radium2 = _interopRequireDefault(_radium);

var _hexCodes = require('./hexCodes');

var _hexCodes2 = _interopRequireDefault(_hexCodes);

var version = '4.4.0';
var url = 'https://maxcdn.bootstrapcdn.com/font-awesome';

var fontFace = '\n  @font-face {\n    font-family: \'FontAwesome\';\n    src: url(\'' + url + '/' + version + '/fonts/fontawesome-webfont.eot?v=' + version + '\');\n    src: url(\'' + url + '/' + version + '/fonts/fontawesome-webfont.eot?#iefix&v=' + version + '\') format(\'embedded-opentype\'),\n      url(\'' + url + '/' + version + '/fonts/fontawesome-webfont.woff2?v=' + version + '\') format(\'woff2\'),\n      url(\'' + url + '/' + version + '/fonts/fontawesome-webfont.woff?v=' + version + '\') format(\'woff\'),\n      url(\'' + url + '/' + version + '/fonts/fontawesome-webfont.ttf?v=' + version + '\') format(\'truetype\'),\n      url(\'' + url + '/' + version + '/fonts/fontawesome-webfont.svg?v=' + version + '#fontawesomeregular\') format(\'svg\');\n    font-weight: normal;\n    font-style: normal;\n  }\n';

var faStyle = {
  display: 'inline-block',
  fontFamily: 'FontAwesome',
  fontSize: 'inherit',
  fontVariant: 'normal',
  fontWeight: 'normal',
  lineHeight: 1,
  textRendering: 'auto',
  WebkitFontSmoothing: 'antialiased',
  MozOsxFontSmoothing: 'grayscale'
};

var fontLoaded = false;

function writeFontFace() {
  var styleEl = document.createElement('style');
  styleEl.setAttribute('type', 'text/css');
  styleEl.innerHTML = fontFace;
  document.head.appendChild(styleEl);
}

var FontAwesome = _react2['default'].createClass({
  displayName: 'FontAwesome',

  propTypes: {
    name: _react.PropTypes.oneOf(_Object$keys(_hexCodes2['default'])).isRequired,
    style: _react.PropTypes.oneOfType([_react.PropTypes.object, _react.PropTypes.array])
  },

  componentWillMount: function componentWillMount() {
    if (!fontLoaded) {
      writeFontFace();
      fontLoaded = true;
    }
  },

  render: function render() {
    var _props = this.props;
    var name = _props.name;
    var style = _props.style;

    var others = _objectWithoutProperties(_props, ['name', 'style']);

    // Get the hex number from the string, then unicode encode it
    var icon = _String$fromCodePoint(parseInt(_hexCodes2['default'][name], 16));
    return _react2['default'].createElement(
      'span',
      _extends({ style: [].concat(faStyle, style) }, others),
      icon
    );
  }
});

exports['default'] = (0, _radium2['default'])(FontAwesome);
module.exports = exports['default'];
